staging/src/k8s.io/kubelet/config/v1beta1/types.go

```diff
@@ -989,6 +989,14 @@ type KubeletConfiguration struct {
 	// +featureGate=GracefulNodeShutdown
 	// +optional
 	ShutdownGracePeriodCriticalPods metav1.Duration `json:"shutdownGracePeriodCriticalPods,omitempty"`
+	// ShutdownGracePeriodByPodPriority specifies the shutdown grace period for Pods based on their associated priority class value.
+	// Each entry in the array represents the graceful shutdown time a pod with a priority class value that
+	// lies in the range of that value and the next higher entry in the list when the node is shutting down.
```

Comment:
There should be a further sentence or two giving an example, for instance:

> For instance, to allow critical pods 90s to shutdown and all remaining pods 300s to shutdown, specify <insert_example_here>.  Pods with priority lower than <value_from_example> will be handled <in this order>.

Reply:
Also, describe the implications of very long grace periods on shutdown in another sentence, maybe:

> The time the Kubelet will wait before exiting will at most be the maximum of all shutdownGracePeriodSeconds for each priority class range represented on the node (<use previous example>).  When all pods have exited or reached their grace periods, the Kubelet will exit.